app/controlplane/internal/conf/controlplane/config/v1/conf.proto

```diff
@@ -135,3 +138,11 @@ message CA {
     string key_pass = 3;
   }
 }
+
+// OnboardingSpec is a configuration to automatically onboard users in organizations with specific roles
+message OnboardingSpec {
+  // name is the name of the organization
+  string name = 1;
+  // role is the role to assign to the user. For the moment, only "viewer" is supported
+  string role = 2;
```

Comment:
what if we use the proto definition we have for a role?

https://github.com/chainloop-dev/chainloop/blob/3f2f0ead388c46c5bdba98a8823832dbb557d5a4/app/controlplane/api/controlplane/v1/response_messages.proto#L145-L149

Then, we can extract this code to be part of the proto package

https://github.com/chainloop-dev/chainloop/blob/3f2f0ead388c46c5bdba98a8823832dbb557d5a4/app/controlplane/internal/service/user.go#L98

something like `(r pb.MembershipRole) ToAuthz ...`

and then we will not need the mapping we do

Reply:
Happy to move the enum as par of the new configuration. The problem comes when extracting that the role mapping, that it creates an importing cycle difficult to solve.
```
2024/06/14 14:33:22 running ent codegen: entc/load: package command-line-arguments
	imports github.com/chainloop-dev/chainloop/app/controlplane/pkg/data/ent/schema
	imports github.com/chainloop-dev/chainloop/app/controlplane/internal/authz
	imports github.com/chainloop-dev/chainloop/app/controlplane/internal/conf/controlplane/config/v1
	imports github.com/chainloop-dev/chainloop/app/controlplane/api/controlplane/v1
	imports github.com/chainloop-dev/chainloop/app/controlplane/internal/authz: import cycle not allowed
```

Reply:
I see, can the mapping be moved to biz? instead of svc? That might help with calling it from the new use-case.

Reply:
Yep, I believe that would do the trick, I'll try it. Thanks.